tools/osbuild-depsolve-dnf

```diff
@@ -17,11 +19,485 @@ import urllib.request
 from datetime import datetime
 from typing import List
 
-import dnf
-import hawkey
+# select "personality" based on argv0
+argv0 = os.path.basename(sys.argv[0]).rsplit("-", 2)[-1]
```

Comment:
This is nice, but... The way we install this is we copy the right script to `/usr/libexec/osbuild-depsolve-dnf` in the spec file. So the actual decision on which one to use is made at build time based on the fedora version and the `images/dnfjson` doesn't need to know anything about dnf5, it just passes the json to whatever script is living there.

Reply:
Thanks, that is very useful to know! I did one more tweak to the PR that switches the way this is detected. Instead of looking at argv0 it now just checks a marker file that the osbuild.spec for fedora 41+ would have to install. Note that this is a strawman, I really don't mind how we signal this, it could be a "ini" file, it could be an env etc. 

If with this change (or a similar one with a different "signaling" way to select the osbuild-depsolve-dnf "personality") the drawbacks are still considered bigger than the benefits I will close this PR (I will be slightly sad though).